Src/Eiffel/eiffel/AST/visitor/ast_value_i_generator.e

```diff
@@ -85,11 +85,13 @@ feature {NONE} -- Implementation
 
 	process_string_as (a_string: STRING_AS)
 		do
-				-- A string constant of the form: a: STRING_8 = "fdsfd"
-				-- does not require any type qualification to the manifest string "fdsfd"
-				-- since the constant type is known to be STRING_8.
-			check type_not_exists: a_string.type = Void end
-			create {STRING_VALUE_I} last_value.make (a_string.value, False)
+				-- A string constant of the form:
+				--  str: STRING_8 = "abc"
+				--  str: STRING_32 = {STRING_32} "abc"
```

Comment:
I do not think we should allow this. The type of the constants is telling us what to do.

Reply:
Ok, my initial idea was to allow s: READABLE_STRING_GENERAL = {IMMUTABLE_STRING_32} "abc" 
but I doubt this is really useful, and for now not implemented this way.
So I will change code, to rely on the constant type declaration, and not rely on typed manifest strings.